mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -753,13 +773,22 @@ else if ( type.getReturnedClass() == ObjectId.class ) {
 		else if ( type instanceof StringAsObjectIdType ) {
 			return StringAsObjectIdGridType.INSTANCE;
 		}
+		else if ( type instanceof BytesAsBsonBinaryType ) {
```

Comment:
I think you can use BinaryType here without the need of this additional class. Unless I'm missing something